src/git.ts

```diff
@@ -108,9 +108,25 @@ export async function getNewCommits(since?: Commit): Promise<CommitSequence> {
     if (since) {
         args.push(headBehindLastMigration ? 'HEAD..' + since.sha1 : since.sha1 + '..HEAD');
     }
-    const [stdout] = await execFile('git', args);
-    const output = stdout.toString().trim();
-    const commits = parseGitLog(output);
+    const commits = await (new Promise<CommitSequence>((resolve, reject) => {
+        const gitProcess = spawn('git', args);
+        let buffer = '';
+        let parsedCommits = new CommitSequence();
+        gitProcess.stdout.on('data', data => {
+            buffer += data.toString().trim();
+            const completeCommits = buffer.substring(0, buffer.lastIndexOf('>>>>COMMIT\n'));
+            buffer = buffer.substring(buffer.lastIndexOf('>>>>COMMIT\n'));
```

Comment:
what if there is no end marker? Seems like this would then cut off one character because lastIndexOf returns -1